meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr-fit/uz/boot.cmd

```diff
@@ -20,15 +20,60 @@ setenv bootloader_s_image ${bootloader_image}
 setenv bootloader2_image "u-boot.itb"
 setenv bootloader2_s_image ${bootloader2_image}
 
-setenv update_primary_image 'echo "${fio_msg} writing ${image_path} ..."; mmc dev ${devnum} && fatwrite mmc ${devnum}:${bootpart} ${loadaddr} boot0001.bin ${filesize}';
-setenv update_secondary_image 'echo "${fio_msg} writing ${image_path} ..."; mmc dev ${devnum} && fatwrite mmc ${devnum}:${bootpart} ${loadaddr} boot0002.bin ${filesize}';
-
-setenv update_primary_image2 'echo "${fio_msg} writing ${image_path} ..."; mmc dev ${devnum} && fatwrite mmc ${devnum}:${bootpart} ${loadaddr} u-boot0001.itb ${filesize}';
-setenv update_secondary_image2 'echo "${fio_msg} writing ${image_path} ..."; mmc dev ${devnum} && fatwrite mmc ${devnum}:${bootpart} ${loadaddr} u-boot0002.itb ${filesize}';
-
 setenv check_board_closed "is_boot_authenticated"
 setenv check_secondary_boot "multi_boot"
-setenv set_primary_boot "multi_boot 1"
-setenv set_secondary_boot "multi_boot 2"
+
+if test "${modeboot}" = "qspiboot"; then
+	# Use SD for open boards, and eMMC for closed
+	run check_board_closed
+
+	if test -n "${board_is_closed}"; then
+		# Use eMMC for further loading/booting Linux FIT image
+		setenv devnum 0
+	else
+		# Use SD for further loading/booting Linux FIT image
+		setenv devnum 1
+	fi
+
+	setenv qspi_bootloader_offset 0x0
+	setenv qspi_bootloader_s_offset 0x60000
+
+	setenv qspi_bootloader2_offset 0x100000
+	setenv qspi_bootloader2_s_offset 0xaa0000
+
+	setenv setup_update 'sf probe && setenv update_cmd "sf update ${loadaddr}"'
+
+	# Boot images (primary and secondary)
+	setenv bootloader_image_update '${qspi_bootloader_offset}'
+	setenv bootloader_s_image_update '${qspi_bootloader_s_offset}'
+
+	# FIT image (primary and secondary)
+	setenv bootloader2_image_update '${qspi_bootloader2_offset}'
+	setenv bootloader2_s_image_update '${qspi_bootloader2_s_offset}'
+
+	setenv set_primary_boot "multi_boot 0"
+	setenv set_secondary_boot "multi_boot 12"
```

Comment:
is this really '12'? sorry, I'm not familiar with multi_boot

Reply:
the layout in QSPI is:

```
0x0             - primary boot.bin
0x60000     - secondary boot.bin
0x100000   - primary u-boot.itb
0xaa0000   - secondary u-boot.itb
```

multiboot represents multiple of `32Kb` (`0x8000`) (as boot image should be placed on 32Kb boundary), so to force BootROM to boot image from `0x60000`, we have to set multiboot register to `0xC` (in dec `12`):
`0x60000 = 0x8000 * 0xC `

Reply:
@igoropaniuk do you mind documenting this somewhere? We will need the required flashing steps and layout for flashing this target.

Reply:
@ricardosalveti makes sense, I'll create a doc for full boot firmware updates for Xilinx ZynqMP SoC (like we have for iMX now, https://docs.foundries.io/latest/reference-manual/security/boot-software-updates.html)